attrd/commands.c

```diff
@@ -713,15 +752,18 @@ attrd_peer_update(crm_node_t *peer, xmlNode *xml, const char *host, bool filter)
         free_xml(sync);
 
     } else if(safe_str_neq(v->current, value)) {
-        crm_info("Setting %s[%s]: %s -> %s from %s", attr, host, v->current, value, peer->uname);
-        free(v->current);
-        if(value) {
-            v->current = strdup(value);
+        if (safe_str_eq(op, ATTRD_OP_UPDATE) || safe_str_eq(op, ATTRD_OP_UPDATE_BOTH)) {
+            crm_info("Setting %s[%s]: %s -> %s from %s", attr, host, v->current, value, peer->uname);
+            free(v->current);
+            if(value) {
+                v->current = strdup(value);
+            } else {
+                v->current = NULL;
+            }
+            changed = TRUE;
         } else {
-            v->current = NULL;
+             crm_trace("Unchanged %s[%s] from %s.(ATTRD_OP_UPDATE_DELAY)", attr, host, peer->uname);
```

Comment:
Hi Ken,

Thank you for comment.

```
(snip)
        if (safe_str_eq(op, ATTRD_OP_UPDATE) || safe_str_eq(op, ATTRD_OP_UPDATE_BOTH)) {
            crm_info("Setting %s[%s]: %s -> %s from %s", attr, host, v->current, value, peer->uname);
            free(v->current);
            if(value) {
                v->current = strdup(value);
            } else {
                v->current = NULL;
            }
            changed = TRUE;
        } else {
             crm_trace("Unchanged %s[%s] from %s.(ATTRD_OP_UPDATE_DELAY)", attr, host, peer->uname);
        }
(snip)
```

Because I have the next problem, I think that this judgment is necessary.

In the case of Y option, the value of the attribute is sent to attrd_peer_update in NULL.
It is not handled in write_or_elect_attribute when a dampen level is the same value.

If there is not this judgment, the attribute level is updated in NULL when I operated the next.

-  I add an attribute by attrd_updater command.
```
[root@rh72-01]# attrd_updater -n test2 -v 5 -d 10

[root@rh72-01]# crm_mon -1 -Af
Stack: corosync
Current DC: rh72-01 (version 1.1.14-e42b4d5) - partition with quorum
Last updated: Tue Mar 15 08:43:06 2016          Last change: Tue Mar 15 08:37:10 2016 by hacluster via crmd on rh72-01

2 nodes and 0 resources configured

Node rh72-02: UNCLEAN (offline)
Online: [ rh72-01 ]


Node Attributes:
* Node rh72-01:
    + test2                             : 5 
(snip)
```
-  I appoint the same value in dampen.
```
[root@rh72-01]# attrd_updater -n test2 -Y -d 10

[root@rh72-01]# crm_mon -1 -Af
Stack: corosync
Current DC: rh72-01 (version 1.1.14-e42b4d5) - partition with quorum
Last updated: Tue Mar 15 08:43:16 2016          Last change: Tue Mar 15 08:37:10 2016 by hacluster via crmd on rh72-01

2 nodes and 0 resources configured

Node rh72-02: UNCLEAN (offline)
Online: [ rh72-01 ]


Node Attributes:
* Node rh72-01:
(snip)
```


What do you think?


Best Regards,
Hideo Yamauchi.